Sara/mkt 774 zero config scanning zcs documentation

s-santillan opened the pull request:
# Thanks for improving Semgrep Docs 😀

### Please ensure

- [x] A subject matter expert (SME) reviews the content
- [x] A technical writer reviews the content or PR
- [x] This change has no security implications or else you have pinged the security team
- [x] Redirects are added if the PR changes page URLs
- [x] If you have changed any header tag links (doc/#this-kind-of-anchor), update all instances of that link

s-santillan commented on docs/deployment/cloud-scanning.md:
```diff
@@ -0,0 +1,182 @@
+
+## Requirements
+
+Cloud scanning requires **[<i class="fas fa-external-link fa-xs"></i> read access](https://docs.github.com/en/rest/authentication/permissions-required-for-github-apps?apiVersion=2022-11-28)** to your code in GitHub for the repositories you choose to scan. Semgrep clones your repository at the beginning of every scan. Once the scan completes, the clone is destroyed and is not persisted anywhere.
```
@jsmoxon Hey Jack! I used the link provided in internal docs, however, it's a list of endpoints - is this intentional? Do we have a more informative link?

jsmoxon replied:
Good question. I see 2 options (open to suggestions):

1. We could link to the GitHub docs on GitHub Apps generally. https://docs.github.com/en/apps/using-github-apps/authorizing-github-apps

2. Or we could link directly to the current link which I believe explains which permissions are granted via the App. I found the current link by following from the GitHub Apps docs and leading towards where they guide you to learn more about permissions. 

```For more information about what different permissions enable a GitHub App to do, see "[Choosing permissions for a GitHub App](https://docs.github.com/en/apps/creating-github-apps/setting-up-a-github-app/choosing-permissions-for-a-github-app).```

@zyannes What do you think makes the most sense?

s-santillan replied:
Hi. My issue with the current link is that it's a list of **possible** permissions we could have for our app, not the actual permission we ask for. Is it **[repository permissions for projects](https://docs.github.com/en/rest/authentication/permissions-required-for-github-apps?apiVersion=2022-11-28#repository-permissions-for-projects)** or is it **[Organization permissions for projects](https://docs.github.com/en/rest/authentication/permissions-required-for-github-apps?apiVersion=2022-11-28#organization-permissions-for-projects)**?

We could also just not have a link. WDYT?

s-santillan replied:
Resolved with link to repository permissions.

s-santillan commented on docs/deployment/cloud-scanning.md:
```diff
@@ -0,0 +1,190 @@
+
+## Scan management and configuration
+
+<!-- I haven't been able to see ANY of these in my ZCS deployment so there are no screenshots. -->
```
None of the buttons (on demand scan through **Run a new scan**, for example) are appearing for me. FYI.

s-santillan commented:
@jsmoxon Btw, I was going to suggest adding a section on how we safeguard the code we have access to (i.e. any relevant legal information?)

jsmoxon commented on docs/deployment/managed-scanning.md:
```diff
@@ -0,0 +1,194 @@
+
+## Feature maturity and support
+
+- Managed scanning is in **public beta** for existing Semgrep customers on a paid plan.
```
@s-santillan This will actually be available to users on the free tier as well.

jsmoxon commented on docs/deployment/managed-scanning.md:
```diff
@@ -0,0 +1,194 @@
+
+### How managed scanning detects repositories
+
+- Repositories with **existing** Semgrep CI jobs are **excluded** from the list. See the following section to convert your existing Semgrep CI jobs to managed scanning.
```
@s-santillan This is no longer the case. Repos with existing Semgrep CI jobs are included in the list. The only thing excluded are repos that are already scanning with managed scanning.

jsmoxon commented on docs/deployment/managed-scanning.md:
```diff
@@ -0,0 +1,194 @@
+- Repositories with **existing** Semgrep CI jobs are **excluded** from the list. See the following section to convert your existing Semgrep CI jobs to managed scanning.
+- Repositories must be accessible to both the public Semgrep GitHub app and the private Semgrep GitHub app.
+
+### Convert or migrate an existing Semgrep CI job
```
@s-santillan I had this wrong originally. It's actually simple to convert to Managed Scanning from CI scans:

1. Onboard the existing project to Semgrep Managed Scanning via the usual method. 
2. Remove the Semgrep.yaml file or cancel the CI job that runs scans in CI. 

That's all you need to do. Deleting a project will actually delete all of the findings so we don't want to encourage that.

pabloest replied:
Does managed scanning pick up any specifics of the CI job settings?

s-santillan replied:
Any specifics: Managed scanning is like a vanilla semgrep.yml file. If you mean user-defined environment variables or additional steps in the yml file to divide a monorepo into submodules, it doesn't include any way to perform those steps. Hence the note on lockfile generation. I made a call out at the top and am wondering if we should call out again separately.

jsmoxon commented on docs/deployment/managed-scanning.md:
```diff
@@ -0,0 +1,194 @@
+1. Select the repositories you want to scan from the list.
+1. Click **Enable managed scanning**.
+
+- After enabling managed scanning, a full scan starts **immediately** on all the repositories you have added.
```
@cgdolan What if a user enables say 100 or 1000 repos? The full scans are not started immediately but threaded out over time, right?

s-santillan replied:
They were indeed threaded for me

jsmoxon reviewed (changes requested):
Generally great @s-santillan. Left a few small comments and 1 larger one about migrating from CI scanning to Managed Scanning.

s-santillan commented:
@jsmoxon Addressed all review notes:

- Renamed "Requirements" to ["Security" and added life cycle tidbit](https://deploy-preview-1536--semgrep-docs-prod.netlify.app/docs/deployment/managed-scanning#security).
- Updated the read access link based on Vivek's note.
- All other comments fixed.

@pabloest 

- Added changes based on your feedback in Slack.

s-santillan commented on docs/deployment/managed-scanning.md:
```diff
@@ -0,0 +1,184 @@
+
+## Requirements
+
+managed scanning requires **read access** to your code in GitHub for the repositories you choose to scan. Semgrep clones your repository at the beginning of every scan. Once the scan completes, the clone is destroyed and is not persisted anywhere.
```
Think this was resolved in a commit whilst you were reviewing.

s-santillan commented on docs/deployment/managed-scanning.md:
```diff
@@ -0,0 +1,184 @@
+
+## Feature maturity and support
+
+- Managed scanning is in **public beta** for all existing Semgrep AppSec Platform users.
```
It's free for everyone :D

s-santillan commented on docs/deployment/deployment-checklist.md:
```diff
@@ -223,6 +239,50 @@ The GitHub integration app is called [`semgrep-app`](https://github.com/apps/sem
+<dt>Writing (and reading) pull requests</dt>
+<dd>Lets Semgrep comment about findings on pull requests.</dd>
+<dt>Writing (and reading) actions</dt>
+<dd>Allows Semgrep AppSec Platform to cancel stuck jobs, rerun jobs, pull logs from jobs, and perform on-demand scanning.</dd>
```
This is taken from the permissions text directly in the product 
![image](https://github.com/semgrep/semgrep-docs/assets/20766840/f3b9f9bd-fc76-49f6-aaa7-864edc967f19)

s-santillan commented:
Added this last tidbit in
![image](https://github.com/semgrep/semgrep-docs/assets/20766840/5b820864-2bbd-4bf0-920d-7edfeaa9c5d0)
